nbd-server.c

```diff
@@ -2514,6 +2614,7 @@ static void handle_request(gpointer data, gpointer user_data) {
 		goto error;
 	}
 
+	pthread_mutex_lock(&package->client->export_lock);
```

Comment:
This is far too coarse locking; essentially, you've turned the server back into a single threaded application.

You need to do this locking in `expread` and `expwrite`, not here. It should preferably also be done in a way that multiple threads can still read and write concurrently, just not (possibly) when we're committing.